### ISSUE TITLE
When opening a long note it takes time to scroll to the bottom

### ISSUE THREAD

**Issue by tomsotte** [standardnotes/mobile#288]
I'm using the v3.0.17 version Android.
I have quite a few long notes where I add some paragraph every now and then all over the day.
Every time I open the note I always have to scroll first to the bottom because the scroll is not restored and I can't find a way to scroll to the top/bottom quickly.

I might find some time to work on a solution for this. I really like the app overall and I'm using it mostly on Android so I'd like to improve this aspects of you consider it a valid improvement.

**Comment by JaspalSuri** [comment]
Hi @tomsotte,

Can you try updating to the latest version of the app via [GitHub](https://github.com/standardnotes/mobile/releases/download/3.0.21/app-release.apk) or the [Play Store](https://play.google.com/store/apps/details?id=com.standardnotes) and let us know if you can still reproduce the issue?

If so, please let us know what editor(s) you're experiencing this issue in/with. 

> Every time I open the note I always have to scroll first to the bottom because the scroll is not restored and I can't find a way to scroll to the top/bottom quickly.

Preserving the scroll position currently isn't possible, but I have taken note of previous requests for this in the past and will add your +1. 🙂 

While it isn't officially supported, I would suggest looking into installing and using the [Append Editor](https://docs.theochu.com/append-editor/#installation) for dealing with large notes, as it provides buttons to jump to the top and bottom of the note. I would be happy to pass along any feedback to the editor's current maintainer. Please note that you will need to install the extension via our web or desktop app first.

**Comment by tomsotte** [comment]
Hi, thanks for the alternative, I've tried the editor and overall I'm ok with it, but on the phone it's kind of clunky and not well integrated. I'm using a dark theme on the app but the editor keep using it's white theme.
Also I did not understand very well why it has to be "hosted" only, on the phone.

**Comment by JaspalSuri** [comment]
> I've tried the editor and overall I'm ok with it, but on the phone it's kind of clunky and not well integrated.

I'm sorry to hear that; I'll pass that info along to the developer. Thank you for trying it out!

> I'm using a dark theme on the app but the editor keep using it's white theme.

Please press and hold on to the theme and select `Redownload` on the menu that appears from below.

> Also I did not understand very well why it has to be "hosted" only, on the phone.

I'm not entirely sure what you mean, but if you're asking why editors are online only - making them available offline is something that's on our long-term roadmap of new features for the mobile app.

Were you also able to update the app with the links that I've provided above? If you've downloaded it from F-Droid, then the version that you have is outdated. The issue with the themes not being downloaded correctly should have been fixed, is why I ask. :-)

**Comment by tomsotte** [comment]
I understand, thanks for having it on the roadmap. Yes I've updated it from the Play Store to have on par with the latest release.
The issue with the theme in particular was with the editor. The theme works correctly on the app shell and on the plain text editor, but the Append editor does not follow the same color palette as the theme.

**Comment by JaspalSuri** [comment]
> I understand, thanks for having it on the roadmap. Yes I've updated it from the Play Store to have on par with the latest release.

Of course! Currently the team is rewriting the apps, which will helps set the foundation for more enhancements in future updates. Thank you for updating the app!

>The issue with the theme in particular was with the editor. The theme works correctly on the app shell and on the plain text editor, but the Append editor does not follow the same color palette as the theme.

Thanks for the feedback! What version of Android are you currently running? I will be sure to pass this along to the developer.

**Comment by tomsotte** [comment]
Hi, sorry for the late response, I'm on Android 10.

**Comment by radko93** [comment]
@tomsotte now hat we refactored apps and 3.5 update is available all themes should be applied to all editors.

**Comment by sidmint** [comment]
Reaching bottom of a long note isn't easy. There should be a way to do that, like in Colornote which I've been using since long, there's a edit icon beside the hamburger menu tapping on which takes one to the bottom while if that is not wanted then one can scroll to any position and tap [double for colornote] there to start editing. Though, tapping edit icon again without exiting a note takes one to the previous editing position. I've many long notes and this makes it so easy.

**Comment by sidmint** [comment]
There can be an alternative by using a scroll bar, just got the idea from elsewhere.
